### PR TITLE
Sync dependencies with main repo and only load required modules

### PR DESCRIPTION
**Pull request by annanay25**
####  Changes
- Removed rust/swift/... compilation tools from `Dockerfile`.
- Synced dependency versions in `build.sh`

Will look into integrating with GitHub Actions to build images and push to dockerhub/quay.io

@yurishkuro - Could you please review?

Signed-off-by: Annanay <annanayagarwal@gmail.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -179,25 +85,18 @@ RUN mkdir -p /upx && curl -sSL https://github.com/upx/upx/releases/download/v${U
...
-COPY --from=rust_builder /out/ /out/
-COPY --from=swift_builder /protoc-gen-swift /out/protoc-gen-swift
 RUN upx --lzma \
         /out/usr/bin/grpc_* \
         /out/usr/bin/protoc-gen-*
 RUN find /out -name "*.a" -delete -or -name "*.la" -delete
 
 FROM alpine:${ALPINE_VERSION}
-LABEL maintainer="Roman Volosatovs <roman@thethingsnetwork.org>"
+LABEL maintainer="Jaegertracing Authors"
```
Comment (yurishkuro) [inline Dockerfile]: We use this for Copyright headers, let's be consistent
```suggestion
LABEL maintainer="The Jaeger Authors"
```

File: build.sh [hunk 2]
```diff
@@ -1,24 +1,12 @@
 #!/usr/bin/env bash
```
Comment (yurishkuro) [inline build.sh]: do we need to keep this build file? I would rather move the versions to Dockerfile, as in https://github.com/znly/docker-protobuf/blob/d22b5e13e718e6378635858e04bd8a6395984dfc/Dockerfile#L4

File: build.sh [hunk 3]
```diff
@@ -1,24 +1,12 @@
...
---build-arg GRPC_RUST_VERSION="${GRPC_RUST_VERSION:-"0.6.2"}" \
---build-arg GRPC_SWIFT_VERSION="${GRPC_SWIFT_VERSION:-"0.10.0"}" \
 --build-arg GRPC_VERSION="${GRPC_VERSION:-"1.26.0"}" \
---build-arg GRPC_WEB_VERSION="${GRPC_WEB_VERSION:-"1.0.7"}" \
---build-arg PROTOBUF_C_VERSION="${PROTOBUF_C_VERSION:-"1.3.2"}" \
---build-arg PROTOC_GEN_DOC_VERSION="${PROTOC_GEN_DOC_VERSION:-"1.3.0"}" \
---build-arg PROTOC_GEN_FIELDMASK_VERSION="${PROTOC_GEN_FIELDMASK_VERSION:-"0.3.3"}" \
---build-arg PROTOC_GEN_GO_VERSION="${PROTOC_GEN_GO_VERSION:-"1.3.3"}" \
+--build-arg PROTOC_GEN_GO_VERSION="${PROTOC_GEN_GO_VERSION:-"1.3.2"}" \
 --build-arg PROTOC_GEN_GOGO_VERSION="${PROTOC_GEN_GOGO_VERSION:-"1.3.1"}" \
```
Comment (yurishkuro) [inline build.sh]: I see this in Jaeger:
```
[[constraint]]
  name = "github.com/gogo/protobuf"
  revision = "ba06b47c162d49f2af050fb4c75bcbc86a159d5c"
```

**Comment by yurishkuro** [comment]
Most importantly, please test this by replacing the proto gen step in jaeger/Makefile with a call to this tool. A successful test would be if we can produce the same `*.pb.go` files as currently exist in Jaeger.

**Comment by annanay25** [comment]
With the latest changes and the following changes to the Jaeger Makefile -

```
diff --git a/Makefile b/Makefile
index 7173bc0..0951ec5 100644
--- a/Makefile
+++ b/Makefile
@@ -418,14 +418,14 @@ generate-mocks: install-mockery
 echo-version:
        @echo $(GIT_CLOSEST_TAG)
 
-PROTOC := protoc
+PROTOC := docker run --rm -v${PWD}:${PWD} -w${PWD} annanay25/jaeger-docker-protobuf:latest -I/usr/include/github.com/gogo/protobuf --proto_path=${PWD}
 PROTO_INCLUDES := \
-       -I model/proto \
-       -I idl/proto \
-       -I vendor/github.com/grpc-ecosystem/grpc-gateway \
-       -I vendor/github.com/gogo/googleapis \
-       -I vendor/github.com/gogo/protobuf/protobuf \
-       -I vendor/github.com/gogo/protobuf
+       -Imodel/proto \
+       -Iidl/proto \
+#      -Ivendor/github.com/grpc-ecosystem/grpc-gateway \
+#      -Ivendor/github.com/gogo/googleapis \
+#      -Ivendor/github.com/gogo/protobuf/protobuf \
+#      -Ivendor/github.com/gogo/protobuf
 # Remapping of std types to gogo types (must not contain spaces)
 PROTO_GOGO_MAPPINGS := $(shell echo \
                Mgoogle/protobuf/descriptor.proto=github.com/gogo/protobuf/types, \
@@ -478,17 +478,17 @@ proto:
 
        $(PROTOC) \
                $(PROTO_INCLUDES) \
-               -I plugin/storage/grpc/proto \
+               -Iplugin/storage/grpc/proto \
                --gogo_out=plugins=grpc,$(PROTO_GOGO_MAPPINGS):$(PWD)/proto-gen/storage_v1 \
                plugin/storage/grpc/proto/storage.proto
 
        $(PROTOC) \
-               -I model/proto \
+               -Imodel/proto \
                --go_out=$(PWD)/model/prototest/ \
                model/proto/model_test.proto
 
        $(PROTOC) \
-               -I plugin/storage/grpc/proto \
+               -Iplugin/storage/grpc/proto \
                --go_out=$(PWD)/plugin/storage/grpc/proto/storageprototest/ \
                plugin/storage/grpc/proto/storage_test.proto
 
```

I was able to generate the exact same files as the repo :)

/cc @yurishkuro

**Comment by annanay25** [comment]
Can we merge this one while we figure out publishing to dockerhub?

**Pull request merged**